api/access.go

```diff
@@ -76,11 +81,131 @@ func Rekey(current, proposed string) error {
 		return err
 	}
 	if err := uri.Post(&respMap, string(b)); err != nil {
-		if rekey_error, present := respMap["error"]; present {
-			return errors.New(rekey_error)
+		if rekeyError, present := respMap["error"]; present {
+			return errors.New(rekeyError)
 		}
 		return err
 	}
 
 	return nil
 }
+
+//Login hits the /v2/auth/login endpoint with the given username and password
+// strings in an attempt to create an authenticated session. Returns a sessionID
+// if successful and an error otherwise. If the error is due to bad credentials,
+// the error returned will be of type ErrUnauthorized.
+func Login(username, password string) (sessionID string, user *AuthIDOutput, err error) {
+	uri, err := ShieldURI("/v2/auth/login")
+	if err != nil {
+		return
+	}
+
+	j, err := json.Marshal(struct {
+		Username string `json:"username"`
+		Password string `json:"password"`
+	}{
+		Username: username,
+		Password: password,
+	})
+
+	if err != nil {
+		panic("Could not marshal auth struct that we JUST made")
```

Comment:
Should we return the error instead of panic()-ing?

Reply:
No. On a json Marshal, the only way that err should occur is if the pointer is nil or if the argument isn't a pointer. Both of those are completely in our control, and this is an assertion that we didn't botch that.